src/services/picture.ts

```diff
@@ -34,4 +38,55 @@ export class PictureService {
       upload.start();
     });
   }
+
+  convertURLtoBlob(URL: string): Promise<Blob> {
+    return new Promise((resolve, reject) => {
+      const image = document.createElement('img');
+
+      image.onload = () => {
+        try {
+          const dataURI = this.convertImageToDataURI(image);
+          const blob = this.convertDataURIToBlob(dataURI);
+
+          resolve(blob);
+        }
+        catch (e) {
+          reject(e);
+        }
+      };
+
+      image.src = URL;
+    });
+  }
+
+  convertImageToDataURI(image: HTMLImageElement): string {
+    // Create an empty canvas element
+    const canvas = document.createElement('canvas');
+    canvas.width = image.width;
+    canvas.height = image.height;
+
+    // Copy the image contents to the canvas
+    const context = canvas.getContext('2d');
+    context.drawImage(image, 0, 0);
+
+    // Get the data-URL formatted image
+    // Firefox supports PNG and JPEG. You could check image.src to
+    // guess the original format, but be aware the using 'image/jpg'
+    // will re-encode the image.
+    const dataURL = canvas.toDataURL('image/png');
+
+    return dataURL.replace(/^data:image\/(png|jpg);base64,/, '');
+  }
+
+  convertDataURIToBlob(dataURI): Blob {
+    const binary = atob(dataURI);
```

Comment:
Same here, you can use a package for that logic: https://www.npmjs.com/package/data-uri-to-blob / https://www.npmjs.com/package/blob-util